app/controllers/posts_controller.rb

```diff
@@ -1,15 +1,50 @@
 # frozen_string_literal: true
 
 class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+
   def new
     @post = Post.new
   end
 
+  # POST /posts
+  # POST /posts.json
   def create
     @post = Post.create(post_params)
     @post.user_name = current_user.name
-    @post.save
-    redirect_to posts_url
+    respond_to do |format|
+      if @post.save
+        format.html { redirect_to posts_url, notice: 'Post was successfully created.' }
+      else
+        format.html { render :new }
+      end
+    end
+  end
+
+  # PATCH/PUT /posts/1
+  # PATCH/PUT /posts/1.json
+  def update
+    respond_to do |format|
+      if @post.update(post_params)
+        format.html { redirect_to posts_url, notice: 'Post was successfully updated.' }
+        format.json { render :show, status: :ok, location: @post }
```

Comment:
format.json is a bit confusing / possibly unnecessary, we should clean that up soon